src/components/messages/conversation-preview.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { FC } from "react";
+import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native";
+import { Conversation, Profile } from "../../models";
+
+const styles = StyleSheet.create({
+  container: {
+    height: 65,
+    justifyContent: "center",
+    marginVertical: 2,
+  },
+  infoContainer: {
+    flexDirection: "row",
+    justifyContent: "space-between",
+    paddingHorizontal: 5,
+  },
+  picture: {
+    height: 60,
+    width: 60,
+    borderRadius: 60 / 2,
+  },
+  titleAndMessageContainer: {
+    flexDirection: "column",
+    alignItems: "flex-start",
+    justifyContent: "center",
+    width: "100%",
+    paddingLeft: 10,
+  },
+  title: {
+    fontWeight: "bold",
+    fontSize: 17,
+    marginBottom: 4,
+  },
+  messageAndTime: {
+    flexDirection: "row",
+    justifyContent: "space-between",
+  },
+  message: {
+    flex: 0.7,
+    fontSize: 14,
+    color: "#9E9E9E",
+  },
+  date: {
+    flex: 0.3,
+    fontSize: 14,
+    color: "#9E9E9E",
+    paddingLeft: 10,
+  },
+});
+
+type Props = {
+  conversation: Conversation;
+  profile: Profile;
+};
+
+export const ConversationPreview: FC<Props> = (props: Props) => {
+  const getTitle: () => string = () => {
+    if (props.conversation.title) return props.conversation.title;
+    const profiles = props.conversation.users.filter(
+      (user) => user.id !== props.profile.id
+    );
+    const fullnames = profiles.map((profile) => {
+      return profile.firstName + " " + profile.lastName;
+    });
+    return fullnames.join(", ");
+  };
+
+  const getPicture: () => string = () => {
+    const profiles = props.conversation.users.filter(
+      (user) => user.id !== props.profile.id
+    );
+    return profiles[0].profilePicture;
+  };
+
+  const getLastMessage: () => string = () => {
+    const lastMessage =
+      props.conversation.messages[props.conversation.messages.length - 1];
+    let lastMessageContent = lastMessage.content;
+
+    if (lastMessage.sender.id === props.profile.id) {
+      lastMessageContent = `Vous : ${lastMessage.content}`;
+    } else if (props.conversation.users.length > 2) {
+      lastMessageContent = `${lastMessage.sender.firstName} : ${lastMessage.content}`;
+    }
+
+    return lastMessageContent;
+  };
+
+  const getTime: () => string = () => {
+    const lastMessage =
+      props.conversation.messages[props.conversation.messages.length - 1];
+    const lastMessageDate = new Date(+lastMessage.timestamp);
+
+    if (Date.now() - lastMessageDate.getTime() < 86400000) {
+      return lastMessageDate
+        .toLocaleTimeString()
+        .split(":")
+        .slice(0, 2)
+        .join(":");
+    }
+
+    return lastMessageDate
+      .toLocaleDateString()
+      .split("/")
+      .slice(0, 2)
+      .join("/");
+  };
```

Comment:
Need feedback on these changes (@Matttx, @Mihobsine)

Reply:
This function also works on my iPhone (with French date format).

But another change that you made is the condition to choose whether to display a time or a date. The previous way was to check if the message was sent the same day and the new way is to check if the last message is more than 24 hours old.
This means, for example, that if a message was sent yesterday à 8 PM if we check the app at 4 PM the next day it will be displayed 20.00 instead of a date.

Reply:
You're right, I didn't think of that! I'll take care of it

Reply:
Should be done @Mistoufle40

Reply:
Yes, it works perfectly and the code is more understandable :)